python/plugins/processing/tools/vector.py

```diff
@@ -111,3 +114,20 @@ def checkMinDistance(point, index, distance, points):
             return False
 
     return True
+
+
+def resolveSourceToFilepath(source):
+    #Tries to resolve a uri-like source (i.e /path/|layername=points) to a valid filepath (/path/points.shp)
+    sourceParts = QgsProviderRegistry.instance().decodeUri('ogr', source)
+    folder = sourceParts.get("path")
+    layerName = sourceParts.get("layerName")
+    if layerName:
+        if os.path.exists(folder):
+            if os.path.isdir(folder):
+                for ext in QgsVectorFileWriter.supportedFormatExtensions():
+                    f = os.path.join(folder, "%s.%s" % (layerName, ext))
```

Comment:
```suggestion
                    f = os.path.join(folder, "{basename}.{extension}".format(basename=layerName, extension=ext))
```

Just because ;)